libsolidity/formal/SMTEncoder.cpp

```diff
@@ -203,7 +203,11 @@ void SMTEncoder::visitFunctionOrModifier()
 		if (dynamic_cast<ContractDefinition const*>(refDecl))
 			visitFunctionOrModifier();
 		else if (auto modifierDef = dynamic_cast<ModifierDefinition const*>(refDecl))
-			inlineModifierInvocation(modifierInvocation.get(), modifierDef);
+		{
+			solAssert(*modifierInvocation->name().annotation().requiredLookup == VirtualLookup::Virtual, "");
```

Comment:
I feel like I already asked this before, but why must it be `virtual`?

Reply:
You did, here is my answer: https://github.com/ethereum/solidity/pull/10614#discussion_r544254357